openlibrary/core/sponsorships.py

```diff
@@ -144,22 +130,26 @@ def qualifies_for_sponsorship(edition, scan_only=False, patron=None):
 
     work_id = work.key.split("/")[-1]
     edition_id = edition.key.split('/')[-1]
-    dwwi, matches = do_we_want_it(edition.isbn, work_id)
+    dwwi, matches = do_we_want_it(edition.isbn)
     if dwwi:
-        bwb_price = get_betterworldbooks_metadata(edition.isbn).get('price_amt')
-        if bwb_price or scan_only:
-            num_pages = int(edition_data['number_of_pages'])
-            scan_price_cents = SETUP_COST_CENTS + (PAGE_COST_CENTS * num_pages)
-            book_cost_cents = int(float(bwb_price) * 100) if not scan_only else 0
-            total_price_cents = scan_price_cents + book_cost_cents
-            resp['price'] = {
-                'book_cost_cents': book_cost_cents,
-                'scan_price_cents': scan_price_cents,
-                'total_price_cents': total_price_cents,
-                'total_price_display': '${:,.2f}'.format(
-                    total_price_cents / 100.
-                ),
-            }
+        num_pages = int(edition_data['number_of_pages'])
+        bwb_price = None
+        if not donate_only:
```

Comment:
This file is currently quite well tested. Let's add some new tests for the new param.